### PR TITLE
Persist onboarding learning times

### LINKED ISSUE #64: Add personal learn time
Wieder in den Einstellungen und gerne auch noch bei in den einzelnen Lernplänen drin.

Image

### PR DESCRIPTION
**Pull request by FleetAdmiralJakob**
## Summary

- derive one canonical `userLearningTimes` record per onboarding weekday from the selected start time and daily study duration
- reject invalid and cross-midnight ranges in the onboarding UI and atomically on the backend, including when Settings already contains an authoritative row
- preserve later Settings edits, close legacy recovery after authoritative changes, and make repeated onboarding synchronization idempotent
- lazily recover only complete, unambiguous, valid legacy onboarding answers without treating those answers as a competing runtime source
- show a brief completion message pointing learners to **Settings → Lernzeiten**
- cover authenticated persistence, tenant isolation, Settings visibility, scheduling consumption, retries, validation, legacy recovery, rollback safety, and rendered correction/completion behavior

## Root cause

Onboarding stored `studyDays`, `learningTime`, and `dailySchoolTime` only as historical `userOnboardingAnswers`. Settings and learning-plan generation read the separate `userLearningTimes` table, so completed onboarding could still leave the learner without scheduling availability.

## User impact

Onboarding-created learning times now appear immediately in Settings and are consumed by learning-plan scheduling. Existing complete and unambiguous legacy answers are recovered safely, while invalid, incomplete, or ambiguous answers continue through the existing setup flow. Once onboarding, legacy recovery, or a Settings change establishes the authoritative schedule, old answers cannot resurrect times the learner later removes.

## Validation

- integrated current `main` (`f89e8e7`) and resolved the onboarding-flow conflict; final effective scope is nine focused files
- `pnpm install --frozen-lockfile` with Node 24.18.0 and pnpm 11.15.1
- `pnpm check` — Biome, ESLint, and TypeScript passing
- `pnpm test` — 70 Vitest files / 352 tests and 21 Jest suites / 52 rendered UI tests passing, plus all auxiliary script tests
- `pnpm format:check`
- `pnpm check:unused`
- `git diff --check`
- independent Standards and Spec reviews — no actionable findings after final re-review
- CodeRabbit hosted review through `3dc149d` — no actionable comments; final hosted incremental review was rate-limited
- CodeRabbit CLI review of the exact final delta (`3dc149d…37e4fd7`) — 0 issues
- GitHub CI replacement run for `37e4fd7` — passed
- all review threads resolved

Linear: [DAY-65](https://linear.app/dayova/issue/DAY-65/add-personal-learn-time)

Closes #64

## UI evidence

https://github.com/user-attachments/assets/122bad5b-09be-47ef-9c18-6b543d9b63fc
> [!NOTE]
I did not re-record it after the latest commits. The existing recording still shows the unchanged onboarding completion UI, while the subsequent changes were backend validation and recovery fixes. It is not evidence from the final commit 37e4fd7.

## Summary by CodeRabbit

- **New Features**
  - Onboarding now validates selected study days, start times, and durations, blocking schedules that would cross midnight.
  - Successfully derived learning times are saved during onboarding and backfilled for returning users when safe.
- **Bug Fixes**
  - Improved onboarding loader completion behavior so navigation to the home screen occurs only when creation is complete.
  - Prevented invalid or ambiguous legacy onboarding data from generating incorrect learning times.
  - Learning time changes in settings now stay authoritative (no unintended resurrection).
- **Tests**
  - Added extensive test coverage for validation, persistence, legacy recovery edge cases, and onboarding UI/error states.

### CODE REVIEW

File: convex/users.ts [hunk 1]
```diff
@@ -177,6 +176,101 @@ const requireIdentity = async (ctx: QueryCtx | MutationCtx) => {
...
+		status: "created" as const,
+		createdCount: derived.windows.length,
+	};
+};
+
+const backfillLegacyLearningTimes = async (
+	ctx: MutationCtx,
+	args: { userId: Id<"users">; ownerTokenIdentifier: string },
+) => {
+	if (await hasLearningTimes(ctx, args.ownerTokenIdentifier)) return;
```
[the remark this answers is not in the document]
Comment (FleetAdmiralJakob) [reply convex/users.ts]: Fixed in `3dc149d`. A monotonic, versioned recovery marker now distinguishes an intentionally empty authoritative schedule from one that has not been migrated. Normal onboarding, terminal/successful legacy recovery, and successful Settings create/update/remove operations establish that marker transactionally. Regression tests cover both onboarding-created and legacy-created schedules being deleted without resurrection, plus the incomplete-legacy Settings path.

File: convex/users.ts [hunk 2]
```diff
@@ -177,6 +176,101 @@ const requireIdentity = async (ctx: QueryCtx | MutationCtx) => {
...
+	ctx: MutationCtx,
+	args: {
+		ownerTokenIdentifier: string;
+		input: OnboardingLearningTimeInput;
+		invalidInput: "reject" | "skip";
+	},
+) => {
+	if (await hasLearningTimes(ctx, args.ownerTokenIdentifier)) {
+		return { status: "preserved" as const, createdCount: 0 };
+	}
```
[the remark this answers is not in the document]
Comment (FleetAdmiralJakob) [reply convex/users.ts]: Fixed in `37e4fd7`. `insertLearningTimesWhenAbsent` now derives and validates the onboarding payload before the preserve-existing branch, so invalid or cross-midnight input aborts atomically even when Settings already contains authoritative rows. The regression test verifies the Settings row remains unchanged and no onboarding answers are persisted.
